apps/api/src/app/events/usecases/map-trigger-recipients/map-trigger-recipients.use-case.ts

```diff
@@ -0,0 +1,139 @@
+  topicId: TopicId;
+  transactionId: string;
+  userId: UserId;
+}
+
+@Injectable()
+export class MapTriggerRecipients {
+  constructor(private createLog: CreateLog, private getTopicSubscribers: GetTopicSubscribersUseCase) {}
+
+  async execute(command: MapTriggerRecipientsCommand): Promise<ISubscribersDefine[]> {
+    const { environmentId, organizationId, recipients, transactionId, userId } = command;
+
+    const mappedRecipients = Array.isArray(recipients) ? recipients : [recipients];
+
+    const simpleSubscribers: ISubscribersDefine[] = this.findSubscribers(mappedRecipients);
+
+    const topicSubscribers: ISubscribersDefine[] = await this.getSubscribersFromAllTopics(
+      transactionId,
+      environmentId,
+      organizationId,
+      userId,
+      mappedRecipients
+    );
+
+    return [...simpleSubscribers, ...topicSubscribers];
+  }
+
+  private async getSubscribersFromAllTopics(
+    transactionId: string,
+    environmentId: EnvironmentId,
+    organizationId: OrganizationId,
+    userId: UserId,
+    recipients: TriggerRecipients
+  ): Promise<ISubscribersDefine[]> {
+    const topics = this.findTopics(recipients);
+
+    const topicSubscribers: ISubscribersDefine[] = [];
+
+    for (const topic of topics) {
+      try {
+        const getTopicSubscribersCommand = GetTopicSubscribersCommand.create({
+          environmentId,
+          topicId: topic?.topicId || '',
+          organizationId,
+          userId,
+        });
+        const response = await this.getTopicSubscribers.execute(getTopicSubscribersCommand);
+        const { subscribers } = response;
+
+        subscribers.forEach((subscriber) => topicSubscribers.push({ subscriberId: subscriber }));
+      } catch (error) {
+        this.logTopicSubscribersError({
+          environmentId,
+          organizationId,
+          topicId: topic?.topicId || '',
+          transactionId,
+          userId,
+        });
+      }
+    }
+
+    return topicSubscribers;
+  }
+
+  public mapSubscriber(subscriber: TriggerRecipientSubscriber): ISubscribersDefine {
+    if (typeof subscriber === 'string') {
+      return { subscriberId: subscriber };
+    }
+
+    return subscriber;
+  }
+
+  private findSubscribers(recipients: TriggerRecipients): ISubscribersDefine[] {
+    const isNotTopic = (recipient: TriggerRecipientSubscriber): recipient is TriggerRecipientSubscriber =>
+      typeof recipient === 'string' || recipient?.type !== TriggerRecipientsTypeEnum.TOPIC;
+
+    return recipients.filter(isNotTopic).map(this.mapSubscriber);
+  }
+
+  private findTopics(recipients: TriggerRecipients): TriggerRecipientTopics {
+    const isTopic = (recipient: ITopic): recipient is ITopic => recipient?.type === TriggerRecipientsTypeEnum.TOPIC;
+
+    return recipients.filter(isTopic);
+  }
+
+  private logTopicSubscribersError({
+    environmentId,
+    organizationId,
+    topicId,
+    transactionId,
+    userId,
+  }: ILogTopicSubscribersPayload) {
+    this.createLog
```

Comment:
I wonder how this will be displayed on the actual new activity feed functionality? I think we can't create an execution detail error here right?

Reply:
Unless I am confused somehow this bit is just to create a log entry.
I think the execution details we should care when we have a plan and a design to show them properly in the Activity Feed, so far every notification generated by a topic _should_ (yet to confirm) be shown as any other one executed.